test/check-machines-lifecycle

```diff
@@ -27,11 +27,16 @@ sys.path.append(os.path.join(TEST_DIR, "common"))
 sys.path.append(os.path.join(os.path.dirname(TEST_DIR), "bots/machine"))
 
 from machineslib import VirtualMachinesCase  # noqa
-from testlib import nondestructive, skipImage, test_main, wait  # noqa
+from testlib import skipImage, test_main, wait  # noqa
+from machine_core.constants import TEST_OS_DEFAULT  # noqa
+from machinesxmls import POOL_XML  # noqa
 
 
-@nondestructive
```

Comment:
This is removed by mistake.

Reply:
It's not. In order to test migration, I need 2 machines for testing to migrate between each other, I need to provision machines. But with nondestructive tests cannot be provisioned. In other words, butting it there would result In this error:

```
# Result testMigrate (__main__.TestMachinesLifecycle) skipped: Cannot provision machines if test is marked as nondestructive
# SKIP testMigrate (__main__.TestMachinesLifecycle) Cannot provision machines if test is marked as nondestructive

```

But now I see that it probably causes the current test failures we see here. So I'm thinking about moving migration to a separate class